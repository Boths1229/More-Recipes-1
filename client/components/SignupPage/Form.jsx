import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  value: PropTypes.shape({
    usernameError: PropTypes.string,
    emailError: PropTypes.string,
    confirmPasswordError: PropTypes.string,
  }),
};

const defaultProps = {
  value: null
};

const Form = props => (
  <div className="container">
    <div className="pt-5" style={{ textAlign: 'center' }}>
      <form className="m-x-auto form card" onSubmit={props.handleSubmit}>
        <h1 className="signin-text">More-Recipes</h1>
        <div className="pb-4 pt-3 form-group">
          <input
            type="text"
            name="username"
            className="form-control"
            required
            onChange={props.onChange}
            onFocus={props.onFocus}
            placeholder="Username"
          />
          {
            props.value.usernameError &&
            <div className="alert alert-danger">
              {
                props.value.usernameError
              }
            </div>
          }
        </div>
        <div className="pb-4 form-group">
          <input
            type="email"
            name="email"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Email"
          />
          {
            props.value.emailError &&
            <div className="alert alert-danger">
              {
                props.value.emailError
              }
            </div>
          }
        </div>
        <div className="pb-4 form-group">
          <input
            type="password"
            id="pass"
            name="password"
            pattern=".{7,12}"
            title="7 to 12 characters"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Password"
          />
        </div>
        <div className="pb-4 form-group">
          <input
            type="password"
            name="confirmPassword"
            className="form-control"
            required
            onFocus={props.onFocus}
            onChange={props.onChange}
            placeholder="Confirm Password"
          />
          {
            props.value.confirmPasswordError &&
            <div className="alert alert-danger">
              {
                props.value.confirmPasswordError
              }
            </div>
          }
        </div>
        <button type="submit" className="btn signin-btn btn-lg" style={{ color: '#f2b43c' }}>
          Create Account
        </button>
        <div className="form-group">
          <p>
            Already have an account? <a href="/signin">Log in.</a>
          </p>
        </div>
      </form>
    </div>
  </div>
);
Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
