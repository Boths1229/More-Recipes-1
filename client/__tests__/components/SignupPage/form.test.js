import React from 'react';
import { shallow } from 'enzyme';
import Form from '../../../components/SignupPage/form';

const value = {
  usernameError: '',
  emailError: '',
  confirmPasswordError: '',
};

let props;

const setup = () => {
  props = {
    value,
    onChange: () => {},
    onFocus: () => {},
    handleSubmit: () => {},
  };

  return shallow(<Form {...props} />);
};


describe('Signup Form component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render error when username error exist', () => {
    const wrapper = setup();
    wrapper.setProps({ value: { usernameError: 'User already exist' } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div.alert').at(0).text()).toBe('User already exist');
  });

  it('should render error when email error exist', () => {
    const wrapper = setup();
    wrapper.setProps({ value: { emailError: 'Email already exist' } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div.alert').at(0).text()).toBe('Email already exist');
  });

  it('should render error when password error exist', () => {
    const wrapper = setup();
    wrapper.setProps({ value: { confirmPasswordError: 'Password does not match' } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div.alert').at(0).text()).toBe('Password does not match');
  });
});
