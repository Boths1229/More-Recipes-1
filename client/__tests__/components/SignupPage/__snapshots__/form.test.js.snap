// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Form component should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="pt-5"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      className="m-x-auto form card"
      onSubmit={[Function]}
    >
      <h1
        className="signin-text"
      >
        More-Recipes
      </h1>
      <div
        className="pb-4 pt-3 form-group"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          id="pass"
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          pattern=".{7,12}"
          placeholder="Password"
          required={true}
          title="7 to 12 characters"
          type="password"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="confirmPassword"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn signin-btn btn-lg"
        style={
          Object {
            "color": "#f2b43c",
          }
        }
        type="submit"
      >
        Create Account
      </button>
      <div
        className="form-group"
      >
        <p>
          Already have an account? 
          <a
            href="/signin"
          >
            Log in.
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;

exports[`Signup Form component should render error when email error exist 1`] = `
<div
  className="container"
>
  <div
    className="pt-5"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      className="m-x-auto form card"
      onSubmit={[Function]}
    >
      <h1
        className="signin-text"
      >
        More-Recipes
      </h1>
      <div
        className="pb-4 pt-3 form-group"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
        <div
          className="alert alert-danger"
        >
          Email already exist
        </div>
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          id="pass"
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          pattern=".{7,12}"
          placeholder="Password"
          required={true}
          title="7 to 12 characters"
          type="password"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="confirmPassword"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn signin-btn btn-lg"
        style={
          Object {
            "color": "#f2b43c",
          }
        }
        type="submit"
      >
        Create Account
      </button>
      <div
        className="form-group"
      >
        <p>
          Already have an account? 
          <a
            href="/signin"
          >
            Log in.
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;

exports[`Signup Form component should render error when password error exist 1`] = `
<div
  className="container"
>
  <div
    className="pt-5"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      className="m-x-auto form card"
      onSubmit={[Function]}
    >
      <h1
        className="signin-text"
      >
        More-Recipes
      </h1>
      <div
        className="pb-4 pt-3 form-group"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          id="pass"
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          pattern=".{7,12}"
          placeholder="Password"
          required={true}
          title="7 to 12 characters"
          type="password"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="confirmPassword"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
        <div
          className="alert alert-danger"
        >
          Password does not match
        </div>
      </div>
      <button
        className="btn signin-btn btn-lg"
        style={
          Object {
            "color": "#f2b43c",
          }
        }
        type="submit"
      >
        Create Account
      </button>
      <div
        className="form-group"
      >
        <p>
          Already have an account? 
          <a
            href="/signin"
          >
            Log in.
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;

exports[`Signup Form component should render error when username error exist 1`] = `
<div
  className="container"
>
  <div
    className="pt-5"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <form
      className="m-x-auto form card"
      onSubmit={[Function]}
    >
      <h1
        className="signin-text"
      >
        More-Recipes
      </h1>
      <div
        className="pb-4 pt-3 form-group"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username"
          required={true}
          type="text"
        />
        <div
          className="alert alert-danger"
        >
          User already exist
        </div>
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          id="pass"
          name="password"
          onChange={[Function]}
          onFocus={[Function]}
          pattern=".{7,12}"
          placeholder="Password"
          required={true}
          title="7 to 12 characters"
          type="password"
        />
      </div>
      <div
        className="pb-4 form-group"
      >
        <input
          className="form-control"
          name="confirmPassword"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn signin-btn btn-lg"
        style={
          Object {
            "color": "#f2b43c",
          }
        }
        type="submit"
      >
        Create Account
      </button>
      <div
        className="form-group"
      >
        <p>
          Already have an account? 
          <a
            href="/signin"
          >
            Log in.
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;
